igvm/commands.py

```diff
@@ -315,6 +323,11 @@ def vm_stop(vm_hostname, force=False):
         else:
             vm.shutdown()
         log.info('"{}" is stopped.'.format(vm.fqdn))
+        if retire:
+            vm.dataset_obj['state'] = 'retired'
+            vm.dataset_obj.commit()
+            log.info('"{}" is retired.'.format(vm.fqdn))
+
```

Comment:
Setting the state before shutdown would make testtool take the VM out of potential LBs before it goes down. Can we move this?

Reply:
Sure, I can do it.

Reply:
It is done.